lightning/src/ln/channelmanager.rs

```diff
@@ -660,6 +660,9 @@ pub(super) enum RAACommitmentOrder {
 // Note this is only exposed in cfg(test):
 pub(super) struct ChannelHolder<Signer: Sign> {
 	pub(super) by_id: HashMap<[u8; 32], Channel<Signer>>,
+	/// SCIDs (and outbound SCID aliases) to the real channel id. Outbound SCID aliases are added
+	/// here once the channel is available for normal use, with SCIDs being added once the funding
+	/// transaction is confirmed at the channel's required confirmation depth.
 	pub(super) short_to_id: HashMap<u64, [u8; 32]>,
```

Comment:
If there is an existing outbound SCID alias, as the routing hop, should we forbid HTLC forward with the _real_ SCID ?

Otherwise a privacy attacker might probe the private channel existence by sending malicious `tlv_payload` built from collecting on-chain p2wsh..

Reply:
That's coming in the next PR - support for the SCIDAlias channel type. We can't do it universally because the channel may be 0conf and public, in which case we want to relay over the real SCID and just use the alias for until we get 6 confs.